### PR TITLE
Biosamtools with libxml2 dep

### PR DESCRIPTION
**Pull request by hajgato**
PR https://github.com/hpcugent/easybuild-easyconfigs/pull/1062 and https://github.com/hpcugent/easybuild-easyconfigs/pull/992 merged... If I did everything right ;)

### CODE REVIEW

**Comment by verdurin** [comment]
I'll take a look at this, given it's been requested by our users this week...

**Comment by boegel** [comment]
@verdurin: you should try and take @hajgato's branch, and update it with develop to start... let me know if you need help with this

**Comment by pescobar** [comment]
Test report by @pescobar
**FAILED**
Build succeeded for 5 out of 71 (15 easyconfigs in this PR)
Linux centos 6.5, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.6.6
See https://gist.github.com/1c9e087f1027800045d6 for a full test report.

**Comment by boegel** [comment]
Jenkins: test this please

**Comment by boegel** [comment]
Test report by @boegel
**FAILED**
Build succeeded for 48 out of 54 (15 easyconfigs in this PR)
Linux SL 6.6, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.6.6
See https://gist.github.com/2fdb480387032bb22c3c for a full test report.

**Comment by boegel** [comment]
@hajgato: can you look into a test report for this?

**Comment by hajgato** [comment]
@boegel: defaults changed since I have did this: it checks for bin, but it does not have.

**Comment by hajgato** [comment]
@boegel sanity_check fixed

**Comment by boegel** [comment]
@pescobar: please retest?

**Comment by boegel** [comment]
@pescobar: hmm, nvm, since this uses rather old toolchains, I'll submit a test report myself

**Comment by boegel** [comment]
Test report by @boegel
**FAILED**
Build succeeded for 11 out of 15 (15 easyconfigs in this PR)
Linux SL 6.6, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.6.6
See https://gist.github.com/aac80dd29f0e66b72aa3 for a full test report.

**Comment by boegel** [comment]
Hmm, failing BioPerl tests:

```
Result: FAIL
Failed 2/334 test programs. 0/19721 subtests failed.
```

I remember looking into this before...

**Comment by hajgato** [comment]
So how do we proceed? For me it was not failed tests. Should I resubmit a test report?

**Comment by boegel** [comment]
@hajgato: if you can submit a test report, please do, yes

I've been trying to find notes or info on the previous time I ran into this, but couldn't find something easily...

**Comment by boegel** [comment]
@hajgato: seems like it may be a missing Perl module, i.e. `XML::LibXML`; maybe you have it on your system?

Adding `XML-LibXML` as a dependency to BioPerl should fix the problem...
```
Bareword found where operator expected at (eval 25) line 2, near "'The optional module XML::LibXML generated the following error: 
Can't"
  (Might be a runaway multi-line '' string starting on line 1)
	(Missing operator before t?)
Backticks found where operator expected at (eval 25) line 2, at end of line
	(Missing semicolon on previous line?)
Undefined subroutine &XML::LibXML::LIBXML_VERSION called at t/SeqIO/seqxml.t line 25.
# Tests were run but no plan was declared and done_testing() was not seen.
# Looks like your test exited with 255 just after 2.
t/SeqIO/seqxml.t ....................... 
```

```
Bareword found where operator expected at (eval 25) line 2, near "'The optional module XML::LibXML generated the following error: 
Can't"
  (Might be a runaway multi-line '' string starting on line 1)
	(Missing operator before t?)
Backticks found where operator expected at (eval 25) line 2, at end of line
	(Missing semicolon on previous line?)
Undefined subroutine &XML::LibXML::LIBXML_VERSION called at t/Tree/TreeIO/phyloxml.t line 21.
# Tests were run but no plan was declared and done_testing() was not seen.
# Looks like your test exited with 255 just after 1.
t/Tree/TreeIO/phyloxml.t ............... 
```

**Comment by hajgato** [comment]
@bogel that was the purpose of this PR, and if you see `XML-LibXML` in this PR. 
I do not know why did not work.

File: easybuild/easyconfigs/b/BioPerl/BioPerl-1.6.924-ictce-5.5.0-Perl-5.20.0.eb [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+perl = 'Perl'
+perlver = '5.20.0'
+versionsuffix = '-%s-%s' % (perl, perlver)
+
+dependencies = [
+    (perl, perlver),
+]
+
+builddependencies = [
+    ('XML-LibXML', '2.0116', versionsuffix),
```
Comment (hajgato) [inline easybuild/easyconfigs/b/BioPerl/BioPerl-1.6.924-ictce-5.5.0-Perl-5.20.0.eb]: @boegel here is the XML-LibXML dep.

File: easybuild/easyconfigs/b/BioPerl/BioPerl-1.6.924-intel-2014b-Perl-5.20.0.eb [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+perl = 'Perl'
+perlver = '5.20.0'
+versionsuffix = '-%s-%s' % (perl, perlver)
+
+dependencies = [
+    (perl, perlver),
+]
+
+builddependencies = [
+    ('XML-LibXML', '2.0116', versionsuffix),
```
Comment (hajgato) [inline easybuild/easyconfigs/b/BioPerl/BioPerl-1.6.924-intel-2014b-Perl-5.20.0.eb]: @boegel here is the XML-LibXML dep

**Comment by hajgato** [comment]
@boegel maybe in the past the builddependency was present in the test pahse, and not anymore?

**Comment by boegel** [comment]
@hajgato: ah, sorry, didn't actually check whether the dep was already there or not, sorry >_<

build dependencies get loaded at the start of the build sequence, and stay there (I think) for the entire process

I'll check on that, and see if I can figure out something on my end.

Meanwhile, can you check whether you have `XML::LibXML` in your system?

I'm getting this:
```
$ perldoc -lm XML::LibXML 
No module found for "XML::LibXML".
```

**Comment by hajgato** [comment]
@boegel Its on it. 
```
# perldoc -lm XML::LibXML 
/usr/lib64/perl5/XML/LibXML.pm
```

**Comment by boegel** [comment]
@hajgato: may be related to https://github.com/hpcugent/easybuild-easyconfigs/issues/643; it's probably not a bad idea to try and figure out why the tests are broken (I see you've disabled them too here)

**Comment by boegel** [comment]
Maybe it has something to do with `-T`? http://perldoc.perl.org/perlsec.html#Taint-mode-and-%40INC

**Comment by migueldiascosta** [comment]
@hajgato, this PR is being closed for the following reason(s): no activity for > 6 months.
Please don't hesitate to reopen this PR or add a comment if you feel this contribution is still relevant.
For more information on our policy w.r.t. closing PRs, see https://easybuild.readthedocs.io/en/latest/Contributing.html#why-a-pull-request-may-be-closed-by-a-maintainer

**Pull request closed without merging**